sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionResolution.scala

```diff
@@ -149,6 +145,10 @@ class FunctionResolution(
           func.prettyName,
           "WITHIN GROUP (ORDER BY ...)"
         )
+      case listAgg: ListAgg
+        if u.isDistinct && !listAgg.isOrderCompatible(u.orderingWithinGroup) =>
```

Comment:
I have two questions:
1. why is this limitation? What's wrong with `listagg(c1) WITHIN GROUP (ORDER BY c2)`?
2. How to make sure the related expressions are fully resolved as we need to do semantic comparison by calling `semanticEquals`?

Reply:
1. Without this check current logic would actually make aggregation over unique pairs (c1, c2). I find this confusing because my intuition says that the `listagg(DISTINCT c1)...` must guarantee the uniqueness of c1. A more suitable solution would be to choose `c2` somehow for every group of `c1` value, but it requires much more effort to implement. And yes, as I mentioned, it's a common limitation.
> E.g. [Snowflake](https://docs.snowflake.com/en/sql-reference/functions/listagg#usage-notes), [DB2](https://www.ibm.com/docs/en/ias?topic=functions-listagg#r0058709__title__4) and Postgress (tested myself)

Reply:
2. Good point. Moved